nixos/modules/services/networking/suricata/default.nix

```
@@ -0,0 +1,222 @@
+{ config
+, pkgs
+, lib
+, ...
+}: let
+  cfg = config.services.suricata;
+  pkg = cfg.package;
+  yaml = pkgs.formats.yaml { };
+  inherit (lib) mkOption types;
+in {
+  meta.maintainers = with lib.maintainers; [ felbinger ];
+
+  options.services.suricata = {
+    enable = lib.mkEnableOption "Suricata";
+
+    package = lib.mkPackageOption pkgs "suricata" { };
+
+    configFile = mkOption {
+      type = types.path;
+      default = pkgs.writeTextFile {
+        name = "suricata.yaml";
+        text = ''
+          %YAML 1.1
+          ---
+          ${builtins.readFile (yaml.generate "suricata-settings-raw.yaml" (lib.filterAttrsRecursive (name: value: value != null) cfg.settings))}
+        '';
+      };
+      description = ''
+        The suricata configuration file, usually you don't want to override the default value
+        and just use the settings options. If you want to include extra configuration to the file
+        use the include option.
```

```suggestion
        Configuration file for suricata.
        
        It is not usual to override the default values; it is recommended to use `settings`. 
        If you want to include extra configuration to the file, use the `includes`.
```

Unfortunately I do not remember the rules for formatting and link referencing inside NixOS modules.